src/odemis/driver/ephemeron.py

```diff
@@ -0,0 +1,1137 @@
+        return state_name
+
+    def read_controller_state(self) -> LocalizedText:
+        """
+        :return: The state of the MightyEBIC controller (see STATE_NAME_*)
+        """
+        f = asyncio.run_coroutine_threadsafe(self._read_controller_state(), self._loop)
+        return f.result()
+
+    async def _read_controller_state(self) -> LocalizedText:
+        ret_val = await self._ebic_state_node.read_value()
+        return ret_val
+
+    def set_controller_state(self, new_state: State):
+        f = asyncio.run_coroutine_threadsafe(self._set_controller_state(new_state), self._loop)
+        f.result()
+
+    async def _set_controller_state(self, new_state: State):
+        await self._ebic_controller_node.call_method(f"{NAMESPACE_INDEX}:set_controller_state", new_state)
+
+    def calculate_scan_time(self, dt: float, p_fast: int, p_slow: int) -> float:
+        f = asyncio.run_coroutine_threadsafe(self._calculate_scan_time(round(dt * 1e6), p_fast, p_slow),
+                                             self._loop)
+        return f.result()
+
+    async def _calculate_scan_time(self, dt: int, p_fast: int, p_slow: int) -> float:
+        """
+        :param dt: dwell time in μs (not "ms" as the documentation claims!)
+        :param p_fast: number of pixels in the fast dimension (X)
+        :param p_slow: number of pixels in the slow dimension (Y)
+        :return: scan time in s
+        """
+        st = await self._ebic_info_node.call_method(f"{NAMESPACE_INDEX}:calculate_scan_time",
+                                                    dt, p_fast, p_slow)
+        return st
+
+    def calculate_dwell_time(self, oversampling: int, channels: int, spp: int, delay: float) -> float:
+        """
+        See _calculate_dwell_time()
+        :return: dwell time in s
+        """
+        f = asyncio.run_coroutine_threadsafe(self._calculate_dwell_time(oversampling, channels, spp, delay),
+                                             self._loop)
+        dt_us = f.result()
+        # If some incorrect paramaters are sent, it returns an int <= 0.
+        if dt_us <= 0:
+            raise ValueError(f"Invalid calculate_dwell_time() returned error {dt_us}.")
+        return dt_us * 1e-6
+
+    async def _calculate_dwell_time(self, oversampling: int, channels: int, spp: int, delay: float) -> int:
+        """
+        Computes how long the measurement of one pixel will take, for the given settings.
+        For the parameters, see _start_trigger_scan().
+        :return: dwell time in μs
+        """
+        if not oversampling in OVERSAMPLING_SET_VALUES:
+            raise ValueError(f"Oversampling value {oversampling} is not valid.")
+        # oversampling, channels, spp, delay -> returns dt (int) in μs
+        dt = await self._ebic_info_node.call_method(
+            f"{NAMESPACE_INDEX}:calculate_dwell_time",
+            oversampling,
+            channels,
+            spp,
+            delay)
```

Comment:
This way of doing multiline parameters, without trailing comma and with closing bracket on same line as last argument, is not the best practice for minimal git diffs (when adding or removing arguments later). However, this approach is prevalent throughout the codebase.